modules/library/api/src/main/java/org/geotools/styling/ContrastEnhancement.java

```diff
@@ -97,31 +86,50 @@
     public Expression getGammaValue();
 
     /**
-     * @deprecated Please use setMethod( ContrastMethodt.NORMALIZE )
+     * Return vendor options relating to the enhancement method
+     * 
+     * @return a Map containing expressions with string keys.
      */
-    @Deprecated
-    public void setNormalize();
-
-     /**
-     * @deprecated Please use setMethod( ContrastMethodt.HISTOGRAM )
+    public Map<String, Expression> getOptions();
+    /**
+     * check if vendor option key is available
+     * @param key - the name of the option to check
+     * @return true if present
+     */
+    public boolean hasOption(String key);
+    
+    /**
+     * Store a vendor option
+     * @param key - the name of the option
+     * @param value an expression that evaluates to it's value
+     */
+    public void addOption(String key, Expression value);
+    
+    /**
+     * Traversal of the style data structure.
+     * @param visitor
      */
-    @Deprecated
-    public void setHistogram();
+    
+    public void accept(StyleVisitor visitor);
 
     /**
-     * @deprecated Please use setMethod; please note Logarithmic is not currently supported
+     * @param options a Map of VendorOptions 
      */
-    @Deprecated
-    public void setLogarithmic();
+    public void setOptions(Map<String, Expression> options);
 
     /**
-     * @deprecated Please use setMethod; please note Exponential is not currently supported
+     * @param string
+     * @return An expression for the matching VendorOption
      */
-    public void setExponential();
+    public Expression getOption(String string);
 
     /**
-     * Traversal of the style data structure.
-     * @param visitor
+     * Convenience method to allow users to pass in a 
+     * {@link ContrastEnhancementMethod} to update {@link Method} 
+     * and {@link Options} in one go.
+     * 
+     * @param method the {@link ContrastEnhancementMethod} that underlies this enhancement
      */
-    public void accept(StyleVisitor visitor);
+    public void setRealMethod(ContrastEnhancementMethod method);
```

Comment:
Rename to setMethod(ContrastEnhancementMethod method) - although you could save syntactic sugar for the builder and leave the API as the data structure.